src/modules/AddOnOthers.rb

```diff
@@ -0,0 +1,79 @@
+# encoding: utf-8
+require "yast"
+require "y2packager/resolvable"
+
+# Yast namespace
+module Yast
+  # This module provides integration of the add-on products
+
+  class AddOnOthersClass < Module
+           
+    include Yast::Logger
+
+    #     add_on_others = [
+    #        {
+    #          "media" => 4, # ID of the source
+    #          "name" : "openSUSE version XX.Y",
+    #          "media_url"=>"dvd:/?devices=/dev/sr1"
+    #          ....
+    #        },
+    #        ...
+    #      ]
+    attr_reader :add_on_others
+
+    def main
+      Yast.import "Pkg"
+      textdomain "packager"      
+
+      @add_on_others = []
+    end
+
+    def Read
+      # Removing all repos which have installed based products
+      # and add-on products.
+      all_products = Y2Packager::Resolvable.find(kind: :product)
+      installed_products = all_products.select do |p|
+        p.status == :available &&
+        all_products.any? { |s| s.name == p.name && s.status == :installed }
+      end
+      installed_src_ids = installed_products.map{ |p| p.source }.uniq
```

Comment:
You can use a Proc symbol here: 
```suggestion
      installed_src_ids = installed_products.map(&:source).uniq
```